otter/supervisor.py

```diff
@@ -108,23 +108,28 @@ def _get_request_bag(self, log, scaling_group):
                                            self.service_configs)
         lb_region = config_value('regionOverrides.cloudLoadBalancers')
 
-        log.msg("Authenticating for tenant")
-        d = self.authenticator.authenticate_tenant(tenant_id, log=log)
-
-        def when_authenticated((auth_token, service_catalog)):
-            bag = RequestBag(
-                lb_region=lb_region or self.region,
-                region=self.region,
-                dispatcher=dispatcher,
-                tenant_id=tenant_id,
-                auth_token=auth_token,
-                service_catalog=service_catalog
-            )
-            return bag
-
-        return d.addCallback(when_authenticated)
-
-    def execute_config(self, log, transaction_id, scaling_group, launch_config):
+        def authenticate():
+            log.msg("Authenticating for tenant")
+            d = self.authenticator.authenticate_tenant(tenant_id, log=log)
+
+            def when_authenticated((auth_token, service_catalog)):
+                bag = RequestBag(
+                    lb_region=lb_region or self.region,
+                    region=self.region,
+                    dispatcher=dispatcher,
+                    tenant_id=tenant_id,
+                    auth_token=auth_token,
+                    service_catalog=service_catalog
+                )
+                bag.re_auth = authenticate
```

Comment:
Could you declare re_auth on RequestBag and then just pass it normally to the constructor instead of assigning it afterwards?